client/human/HumanClientApp.cpp

```diff
@@ -1238,3 +1243,42 @@ void HumanClientApp::OpenURL(const std::string& url) {
     // execute open command
     system(command.c_str());
 }
+
+void HumanClientApp::AddWaitingFlag(std::string reason) {
+    if (reason.empty()) {
+        ErrorLogger() << "Requested addition of empty wait flag";
+        return;
+    }
+
+    TraceLogger() << "Request to add wait flag: " << reason;
+    m_wait_flags[reason]++;
```

Comment:
@geoffthemedio Good point.
I am guessing that adding functionality to GG instead and using signal/slots would be preferred (and provide a better avenue for other cursor changes).
I will look into making such a change unless another avenue is suggested, I need to study up on GG::GUIImpl/atomic/SDL to determine how to approach.
Will close this PR as invalid and open as issue if that sounds like the correct course.